Tests/MarkdownTests/Base/MarkupTests.swift

```diff
@@ -283,4 +283,79 @@ final class MarkupTests: XCTestCase {
             )!.debugDescription()
         )
     }
+    
+    func testChildAtPositionHasCorrectType() throws {
```

Comment:
This is a nice test addition, thanks. Please also add some checks that children fetched with `child(at:)` have the expected `self._data.id`.

Reply:
Addressed here: 6465f073530fc77714edad9c004f5d26a5fea622.

Reply:
Excellent.